frog/imports/client/UserDashboard/components/ArchivesPage/index.js

```diff
@@ -0,0 +1,179 @@
+                      itemIcon={itemIcon}
+                      itemType={itemType}
+                      dateCreated={dateCreated}
+                      callback={callback}
+                      secondaryActions={secondaryActions}
+                    />
+                  );
+                }
+              )}
+            </List>
+          </Paper>
+        </Grid>
+      )}
+      {archivesSessions.length > 0 && (
+        <Grid
+          item
+          xs={
+            archivesDrafts.length > 0 || archivesTemplates.length > 0 ? 6 : 12
+          }
+        >
+          <Paper className={classes.paper} elevation={0}>
+            <Typography variant="h5">
+              <Bookmark /> Sessions
+            </Typography>
+
+            <List dense>
+              {archivesSessions.map(
+                (
+                  {
+                    itemIcon,
+                    itemTitle,
+                    status,
+                    itemType,
+                    dateCreated,
+                    callback,
+                    secondaryActions
+                  },
+                  index
+                ) => (
+                  <ContentListItem
+                    key={index}
+                    itemIcon={itemIcon}
+                    itemTitle={itemTitle}
+                    itemType={itemType}
+                    dateCreated={dateCreated}
+                    status={status}
+                    callback={callback}
+                    secondaryActions={secondaryActions}
+                  />
+                )
+              )}
+            </List>
+          </Paper>
+        </Grid>
+      )}
+      {archivesTemplates.length > 0 && (
+        <Grid
+          item
+          xs={archivesSessions.length > 0 || archivesDrafts.length > 0 ? 6 : 12}
+        >
+          <Paper className={classes.paper} elevation={0}>
+            <div className={classes.buttonRows}>
+              <Typography variant="h5" align="left">
+                <DescriptionIcon /> Templates
+              </Typography>
+            </div>
+
+            <List>
+              {archivesTemplates.map(
+                (
+                  {
+                    itemIcon,
+                    itemTitle,
+                    itemType,
+                    dateCreated,
+                    callback,
+                    secondaryActions
+                  },
+                  index
+                ) => {
+                  return (
+                    <ContentListItem
+                      key={index}
+                      itemTitle={itemTitle}
+                      itemIcon={itemIcon}
+                      itemType={itemType}
+                      dateCreated={dateCreated}
+                      callback={callback}
+                      secondaryActions={secondaryActions}
+                    />
+                  );
+                }
+              )}
+            </List>
+          </Paper>
+        </Grid>
+      )}
+    </Grid>
+  );
+};
```

Comment:
I don't like this file much. It seems to me that it is a near identical copy-paste of the `RecentPage` component. For code quality we should avoid such things. Do you see a way to reuse the same component instead of duplicating the code?

Reply:
The only difference between this and the `RecentPage` component is the display limit of 6 data rows. We can definitely add a variant in the RecentPage to accommodate this but my reasoning for keeping the two components separate was the future scope of expanding the RecentPage to add more functionality.